modules/mapbox/src/deck-utils.js

```diff
@@ -28,7 +33,8 @@ export function getDeckInstance({map, gl, deck}) {
     userData: {
       isExternal: false,
       mapboxLayers: new Set()
-    }
+    },
+    views: deck.props.views || [new MapView({id: 'mapbox'})]
```

Comment:
This will be overwritten if you use React and the component rerenders. `views` must be resolved before `drawLayer` is called, not at initialization.

Reply:
Should I memoize `new MapView({id: 'mapbox'})` in userData, or is it fine to re-initialize on drawLayer calls?

Reply:
You don't need to mutate the prop at all. In applyViewport you can just use the viewport ids to figure out where to position the map viewport.

Reply:
That would work if the "mapbox" view is present, right? Would we insert the view if it's missing somewhere?

Reply:
You don't need the view since deck is not providing the controller here.

Reply:
Ok, I see two `_drawLayers` calls but it isn't clear how to only manipulate views for external deck instances. 

In `afterRender`, `applyViewport` is only called when deck is external - that is clear. But `drawLayer` it is called as long as `deck.layerManager` is defined - there is no check for deck being external, so it is unclear.